src/structure/model.jl

```diff
@@ -0,0 +1,1089 @@
+atomname(atom::Atom) = atom.name
+
+"Get the alternative location ID of an `AbstractAtom`."
+altlocid(atom::Atom) = atom.alt_loc_id
+
+"Get the residue name of an `AbstractAtom` or `AbstractResidue`."
+resname(atom::Atom) = atom.res_name
+
+"Get the chain ID of an `AbstractAtom`, `AbstractResidue` or `Chain`."
+chainid(atom::Atom) = atom.chain_id
+
+"Get the residue number of an `AbstractAtom` or `AbstractResidue`."
+resnumber(atom::Atom) = atom.res_number
+
+"Get the insertion code of an `AbstractAtom` or `AbstractResidue`."
+inscode(atom::Atom) = atom.ins_code
+
+"Get the x coordinate of an `AbstractAtom`."
+x(atom::Atom) = atom.coords[1]
+
+"Get the y coordinate of an `AbstractAtom`."
+y(atom::Atom) = atom.coords[2]
+
+"Get the z coordinate of an `AbstractAtom`."
+z(atom::Atom) = atom.coords[3]
+
+"Get the atomic coordinates of an `AbstractAtom` as a `Vector{Float64}`."
+coords(atom::Atom) = atom.coords
+
+"""
+Get the occupancy of an `AbstractAtom`. Defaults to 1.0 if not read from the PDB
+file.
+"""
+occupancy(atom::Atom) = atom.occupancy
+
+"""
+Get the temperature factor on an `AbstractAtom`. Defaults to 0.0 if not read
+from the PDB file.
+"""
+tempfac(atom::Atom) = atom.temp_fac
+
+"""
+Get the element of an `AbstractAtom`. Defaults to \"\" if not read from the PDB
+file.
+"""
+element(atom::Atom) = atom.element
+
+"""
+Get the charge on an `AbstractAtom`. Defaults to \"\" if not read from the PDB
+file.
+"""
+charge(atom::Atom) = atom.charge
+
+
+"""
+Determines if an `AbstractAtom` represents a hetero atom, i.e. came from a
+HETATM record, or if an `AbstractResidue` represents a hetero molecule, i.e.
+consists of HETATM records.
+"""
+ishetero(atom::AbstractAtom) = ishetatom(atom)
+
+"""
+Determines if an `AbstractAtom` is a `DisorderedAtom`, i.e. if there are
+multiple locations present for an atom.
+"""
+isdisorderedatom(::Atom) = false
+
+"""
+Get a descriptive atom ID for an `AbstractAtom` as a `Tuple` of the form
+(full residue ID, residue name, atom name).
+"""
+atomid(atom::Atom) = (resid(atom; full=true), resname(atom), atomname(atom))
+
+
+"""
+Set the x coordinate of an `AbstractAtom`. For `DisorderedAtom`s only the
+default atom is updated.
+"""
+x!(atom::Atom, x::Real) = (atom.coords[1] = x; nothing)
+
+"""
+Set the y coordinate of an `AbstractAtom`. For `DisorderedAtom`s only the
+default atom is updated.
+"""
+y!(atom::Atom, y::Real) = (atom.coords[2] = y; nothing)
+
+"""
+Set the z coordinate of an `AbstractAtom`. For `DisorderedAtom`s only the
+default atom is updated.
+"""
+z!(atom::Atom, z::Real) = (atom.coords[3] = z; nothing)
+
+
+"""
+Set the coordinates of an `AbstractAtom` to a `Vector{Float64}` of length 3. For
+`DisorderedAtom`s only the default atom is updated.
+"""
+function coords!(atom::Atom, coords::Vector{Float64})
+    @assert length(coords) == 3 "3 coordinates must be given"
+    x!(atom, coords[1])
+    y!(atom, coords[2])
+    z!(atom, coords[3])
+    return nothing
+end
```

Comment:
These functions (`x!`, `y!`, `z!`, and `coords!`) also should return the modified object (i.e. `atom`).